utils.h

```diff
@@ -176,7 +175,12 @@ static inline int is_cue_url(const char *name)
 
 static inline int is_url(const char *name)
 {
-	return is_http_url(name) || is_cdda_url(name) || is_cue_url(name);
+	return is_http_url(name) || is_cdda_url(name);
+}
+
+static inline int is_partial_url(const char *name)
```

Comment:
Yes, I do not like these functions either; I'd even say that their existence makes this PR feel unfinished.